File: src/Entity/Echeance.php
<?php

namespace App\Entity;

use App\Repository\EcheanceRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: EcheanceRepository::class)]
class Echeance
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(inversedBy: 'echeances')]
    private ?Credit $id_credit = null;

    #[ORM\Column(length: 255)]
    private ?string $mode_paiement = null;

    #[ORM\Column(length: 255)]
    private ?string $etat = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE, nullable: true)]
    private ?\DateTimeInterface $date = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getIdCredit(): ?Credit
    {
        return $this->id_credit;
    }

    public function setIdCredit(?Credit $id_credit): static
    {
        $this->id_credit = $id_credit;

        return $this;
    }

    public function getModePaiement(): ?string
    {
        return $this->mode_paiement;
    }

    public function setModePaiement(string $mode_paiement): static
    {
        $this->mode_paiement = $mode_paiement;

        return $this;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): static
    {
        $this->etat = $etat;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(?\DateTimeInterface $date): static
    {
        $this->date = $date;

        return $this;
    }
}

File: src/main/java/com/inventorymanagementsystem/entity/Echeance.java
package com.inventorymanagementsystem.entity;

import java.sql.Date;

public class Echeance {
    private int id, idCredit;
    private String ModePyament,Etat;
    private Date Date;

    public Echeance() {
    }

    public Echeance(int id, int idCredit, String modePyament, String etat, java.sql.Date date) {
        this.id = id;
        this.idCredit = idCredit;
        ModePyament = modePyament;
        Etat = etat;
        Date = date;
    }
    public Echeance(int idCredit, String modePyament, String etat, java.sql.Date date) {
        this.idCredit = idCredit;
        ModePyament = modePyament;
        Etat = etat;
        Date = date;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getIdCredit() {
        return idCredit;
    }

    public void setIdCredit(int idCredit) {
        this.idCredit = idCredit;
    }

    public String getModePyament() {
        return ModePyament;
    }

    public void setModePyament(String modePyament) {
        ModePyament = modePyament;
    }

    public String getEtat() {
        return Etat;
    }

    public void setEtat(String etat) {
        Etat = etat;
    }

    public java.sql.Date getDate() {
        return Date;
    }

    public void setDate(java.sql.Date date) {
        Date = date;
    }

    @Override
    public String toString() {
        return "Echeance{" +
                "id=" + id +
                ", idCredit=" + idCredit +
                ", ModePyament='" + ModePyament + '\'' +
                ", Etat='" + Etat + '\'' +
                ", Date=" + Date +
                '}';
    }
}
